pkg/pool-stable/contracts/ComposableStablePool.sol

```diff
@@ -678,6 +682,35 @@ contract ComposableStablePool is
         return (bptAmount, _addBptItem(amountsDelta, 0));
     }
 
+    /**
+     * @dev Pay any due protocol fees and calculate values necessary for performing the join/exit.
+     */
+    function _beforeJoinExit(uint256[] memory registeredBalances)
+        internal
+        returns (
+            uint256,
+            uint256[] memory,
+            uint256,
+            uint256
+        )
+    {
+        (uint256 lastJoinExitAmp, uint256 lastPostJoinExitInvariant) = getLastJoinExitData();
+        (
+            uint256 preJoinExitSupply,
+            uint256[] memory balances,
+            uint256 oldAmpInvariant
+        ) = _payProtocolFeesBeforeJoinExit(registeredBalances, lastJoinExitAmp, lastPostJoinExitInvariant);
+        (uint256 currentAmp, ) = _getAmplificationParameter();
+
+        // If the amplification factor is the same as it was during the last join/exit then we can reuse the
+        // value calculated using the "old" amplification factor. If not, then we have to calculate this now.
+        uint256 preJoinExitInvariant = currentAmp != lastJoinExitAmp
```

Comment:
```suggestion

        (
            uint256 preJoinExitSupply,
            uint256[] memory balances,
            uint256 oldAmpInvariant
        ) = _payProtocolFeesBeforeJoinExit(registeredBalances, lastJoinExitAmp, lastPostJoinExitInvariant);

        // If the amplification factor is the same as it was during the last join/exit then we can reuse the
        // value calculated using the "old" amplification factor. If not, then we have to calculate this now.
        (uint256 currentAmp, ) = _getAmplificationParameter();
        uint256 preJoinExitInvariant = currentAmp != lastJoinExitAmp
```

Nitpicky - the original whitespace made it a bit hard for me to understand what was going on.